elephant/joint_decorrelation/joint_decorrelation.py

```diff
@@ -0,0 +1,139 @@
+import numpy as np
+import neo
+import quantities as pq
+from meegkit.dss import dss0
+
+class JD():
+    def __init__(self, n_max_components=None, explained_variance_threshold=1e-9):
+        """
+        n_max_components: int
+            Number of PCs to retain (default=all).
+        explained_variance_threshold: float
+            Ignore PCs smaller than explained_variance_threshold (default=10.^-9).
+        """
+        self.n_max_components = n_max_components
+        self.explained_variance_threshold = explained_variance_threshold
+
+    def fit(self, baseline_data=None, bias_filtered_data=None,
+            baseline_covmat=None, bias_filtered_covmat=None, verbose=False):
+
+        # check data shape compatibility
+        if baseline_covmat is None and bias_filtered_covmat is None:
+            if baseline_data.shape[0] != bias_filtered_data.shape[0]:
+                raise ValueError('`baseline_data` and `bias_filtered_data`'
+                                 'should have same size in the first dimension.')
+        else:
+            if not (baseline_covmat.shape[0] == baseline_covmat.T.shape[1] and
+                    np.all(np.ravel(baseline_covmat) == np.ravel(baseline_covmat.T))):
+                raise ValueError('`baseline_covmat` should be a symmetric square'
+                                 'matrix.')
+            if not (bias_filtered_covmat.shape[0] == bias_filtered_covmat.T.shape[1] and
+                    np.all(np.ravel(bias_filtered_covmat) == np.ravel(bias_filtered_covmat.T))):
+                raise ValueError('`bias_filtered_covmat` should be a symmetric square'
+                                 'matrix.')
+            # check data shape compatibility
+            if baseline_covmat.shape != bias_filtered_covmat.shape:
+                raise ValueError('`baseline_covmat` and `bias_filtered_covmat` should have same'
+                                 'size.')
+
+        if baseline_covmat is None and bias_filtered_covmat is None:
+            c0 = np.cov(baseline_data)
+            c1 = np.cov(bias_filtered_data)
+        else:
+            c0 = baseline_covmat
+            c1 = bias_filtered_covmat
+
+        return self._fit(c0, c1, verbose)
+
+    def _fit(self, c0, c1, verbose):
+
+        todss, fromdss, pwr0, pwr1 = dss0(c0, c1, keep1=self.n_max_components,
```

Comment:
these only-god-knows variables probably come from a paper
please specify this in the class doc. maybe add formulas too